```
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_basic_math.cpp:12:
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/test/doc/explicit_use_of_an_instruction_set_mean_tag_dispatch.cpp:1:
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_batch.cpp:12:
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:104: test/CMakeFiles/test_xsimd.dir/test_arch.cpp.o] Error 1
xsimd-x86_64-darwin> make[2]: *** Waiting for unfinished jobs....
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_batch_bool.cpp:12:
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/test/test_batch_cast.cpp:12:
xsimd-x86_64-darwin> In file included from /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/xsimd.hpp:55:
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:483:16: error: use of undeclared identifier '__exp10f'
xsimd-x86_64-darwin>         return __exp10f(x);
xsimd-x86_64-darwin>                ^
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:487:16: error: use of undeclared identifier '__exp10'; did you mean 'exp10'?
xsimd-x86_64-darwin>         return __exp10(x);
xsimd-x86_64-darwin>                ^~~~~~~
xsimd-x86_64-darwin>                exp10
xsimd-x86_64-darwin> /tmp/nix-build-xsimd-11.1.0.drv-0/source/include/xsimd/arch/xsimd_scalar.hpp:485:19: note: 'exp10' declared here
xsimd-x86_64-darwin>     inline double exp10(const double& x) noexcept
xsimd-x86_64-darwin>                   ^
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:90: test/CMakeFiles/test_xsimd.dir/test_api.cpp.o] Error 1
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:118: test/CMakeFiles/test_xsimd.dir/test_basic_math.cpp.o] Error 1
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:160: test/CMakeFiles/test_xsimd.dir/test_batch_cast.cpp.o] Error 1
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/doc/CMakeFiles/test_doc_any_arch.dir/build.make:118: test/doc/CMakeFiles/test_doc_any_arch.dir/explicit_use_of_an_instruction_set_mean_tag_dispatch.cpp.o] Error 1
xsimd-x86_64-darwin> make[1]: *** [CMakeFiles/Makefile2:175: test/doc/CMakeFiles/test_doc_any_arch.dir/all] Error 2
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:146: test/CMakeFiles/test_xsimd.dir/test_batch_bool.cpp.o] Error 1
xsimd-x86_64-darwin> 2 errors generated.
xsimd-x86_64-darwin> make[2]: *** [test/CMakeFiles/test_xsimd.dir/build.make:132: test/CMakeFiles/test_xsimd.dir/test_batch.cpp.o] Error 1
xsimd-x86_64-darwin> make[1]: *** [CMakeFiles/Makefile2:120: test/CMakeFiles/test_xsimd.dir/all] Error 2
xsimd-x86_64-darwin> make: *** [Makefile:146: all] Error 2

--- a/pkgs/development/libraries/xsimd/default.nix
+++ b/pkgs/development/libraries/xsimd/default.nix
@@ -1,4 +1,10 @@
-{ lib, stdenv, fetchFromGitHub, cmake, doctest }:
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, doctest
+}:
+
 stdenv.mkDerivation rec {
   pname = "xsimd";
   version = "11.1.0";
@@ -9,14 +15,18 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-l6IRzndjb95hIcFCCm8zmlNHWtKduqy2t/oml/9Xp+w=";
   };
 
-  nativeBuildInputs = [ cmake ];
+  nativeBuildInputs = [
+    cmake
+  ];
 
   cmakeFlags = [
     "-DBUILD_TESTS=${if (doCheck && stdenv.hostPlatform == stdenv.buildPlatform) then "ON" else "OFF"}"
   ];
 
   doCheck = true;
-  nativeCheckInputs = [ doctest ];
+  nativeCheckInputs = [
+    doctest
+  ];
   checkTarget = "xtest";
 
   meta = with lib; {

--- a/pkgs/development/libraries/xsimd/default.nix
+++ b/pkgs/development/libraries/xsimd/default.nix
@@ -14,6 +14,18 @@ stdenv.mkDerivation rec {
     rev = version;
     sha256 = "sha256-l6IRzndjb95hIcFCCm8zmlNHWtKduqy2t/oml/9Xp+w=";
   };
+  patches = [
+    # Ideally, Accelerate/Accelerate.h should be used for this implementation,
+    # but it doesn't work... Needs a Darwin user to debug this. We apply this
+    # patch unconditionally, because the #if macros make sure it doesn't
+    # interfer with the Linux implementations.
+    ./fix-darwin-exp10-implementation.patch
+  ] ++ lib.optionals stdenv.isDarwin [
+    # Upstream reports:
+    # https://github.com/xtensor-stack/xsimd/issues/807
+    # https://github.com/xtensor-stack/xsimd/issues/917
+    ./disable-darwin-failing-tests.patch
+  ];
 
   nativeBuildInputs = [
     cmake

--- a/pkgs/development/libraries/xsimd/disable-darwin-failing-tests.patch
+++ b/pkgs/development/libraries/xsimd/disable-darwin-failing-tests.patch
@@ -1,0 +1,58 @@
+diff --git i/test/test_error_gamma.cpp w/test/test_error_gamma.cpp
+index 214cbb5..299e5b8 100644
+--- i/test/test_error_gamma.cpp
++++ w/test/test_error_gamma.cpp
+@@ -131,25 +131,6 @@ struct error_gamma_test
+             INFO("lgamma");
+             CHECK_EQ(diff, 0);
+         }
+-#if !(XSIMD_WITH_AVX && !XSIMD_WITH_AVX2)
+-
+-        // tgamma (negative input)
+-        {
+-            std::transform(gamma_neg_input.cbegin(), gamma_neg_input.cend(), expected.begin(),
+-                           [](const value_type& v)
+-                           { return std::lgamma(v); });
+-            batch_type in, out;
+-            for (size_t i = 0; i < nb_input; i += size)
+-            {
+-                detail::load_batch(in, gamma_neg_input, i);
+-                out = lgamma(in);
+-                detail::store_batch(out, res, i);
+-            }
+-            size_t diff = detail::get_nb_diff(res, expected);
+-            INFO("lgamma (negative input)");
+-            CHECK_EQ(diff, 0);
+-        }
+-#endif
+     }
+ };
+ 
+diff --git i/test/test_xsimd_api.cpp w/test/test_xsimd_api.cpp
+index 84b4b0b..1b29742 100644
+--- i/test/test_xsimd_api.cpp
++++ w/test/test_xsimd_api.cpp
+@@ -515,11 +515,6 @@ struct xsimd_api_float_types_functions
+         value_type val(2);
+         CHECK_EQ(extract(xsimd::exp(T(val))), std::exp(val));
+     }
+-    void test_exp10()
+-    {
+-        value_type val(2);
+-        CHECK_EQ(extract(xsimd::exp10(T(val))), std::pow(value_type(10), val));
+-    }
+     void test_exp2()
+     {
+         value_type val(2);
+@@ -804,11 +799,6 @@ TEST_CASE_TEMPLATE("[xsimd api | float types functions]", B, FLOAT_TYPES)
+         Test.test_exp();
+     }
+ 
+-    SUBCASE("exp10")
+-    {
+-        Test.test_exp10();
+-    }
+-
+     SUBCASE("exp2")
+     {
+         Test.test_exp2();

--- a/pkgs/development/libraries/xsimd/fix-darwin-exp10-implementation.patch
+++ b/pkgs/development/libraries/xsimd/fix-darwin-exp10-implementation.patch
@@ -1,0 +1,22 @@
+diff --git i/include/xsimd/arch/xsimd_scalar.hpp w/include/xsimd/arch/xsimd_scalar.hpp
+index 9066da6..7aa3b6b 100644
+--- i/include/xsimd/arch/xsimd_scalar.hpp
++++ w/include/xsimd/arch/xsimd_scalar.hpp
+@@ -502,16 +502,7 @@ namespace xsimd
+         return !(x0 == x1);
+     }
+ 
+-#if defined(__APPLE__)
+-    inline float exp10(const float& x) noexcept
+-    {
+-        return __exp10f(x);
+-    }
+-    inline double exp10(const double& x) noexcept
+-    {
+-        return __exp10(x);
+-    }
+-#elif defined(__GLIBC__)
++#if defined(__GLIBC__)
+     inline float exp10(const float& x) noexcept
+     {
+         return ::exp10f(x);

